.ci/generate.jenkinsfile

```diff
@@ -352,15 +352,20 @@ node('Linux') {
                 withCredentials([usernamePassword(credentialsId: 'docker-credentials', usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]) {
                     sh 'docker login --username $USERNAME --password $PASSWORD'
                 }
+                String tagVersion = params.conan_version
+                String parsedVersion = parseVersion(params.conan_version)
+                if (parsedVersion[0] == 2) {
+                    tagVersion = "2.0.0-pre"
+                }
 
                 // Upload with the corresponding tag version
-                String uploadImage = "${dockerhubUsername}/${image}:${params.conan_version}"
+                String uploadImage = "${dockerhubUsername}/${image}:${tagVersion}"
                 sh "docker tag ${builtImage} ${uploadImage}"
                 if (!params.dry_run) {
                     sh "docker push ${uploadImage}"
                 }
 
-                if (params.upload_latest) {
+                if (params.upload_latest && parsedVersion[0] == 1) {
```

Comment:
This should be handled somewhere else, in the caller, by providing `params.upload_latest=false`, but it might be good to have this extra check here